apps/sandbox/src/hooks/useWallet.ts

```diff
@@ -0,0 +1,74 @@
+import { useWalletState } from '@bos-web-engine/wallet-selector-control';
```

Comment:
This is a totally fair approach as well - @mpeterdev and I ran through this idea too. I'm happy to refactor all of the `useWallet()` boilerplate into a single like your example shows. I'll do that now.

Reply:
Hey @andy-haynes, I did a full on refactor of how `useWallet()` works. I'm now using a more a standard provider/context pattern to initialize the wallet selector instance. Then `useWallet()` simply pulls in that context. This will provide more stability when sharing the wallet selector instance - and make it clearer how to initialize the wallet selector with less boilerplate.

Reply:
I like the new approach! Nice work covering the complexity with managing instances.